crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/cheat_execution_info.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    state::{CheatSpan, CheatStatus},
+    CheatnetState,
+};
+use cairo_felt::Felt252;
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+
+#[derive(Clone, Debug)]
+pub struct CheatArguments<T> {
+    value: T,
+    span: CheatSpan,
+    target: ContractAddress,
+}
+
+#[derive(Clone, Default, Debug)]
+pub enum Operation<T> {
+    StartGlobal(T),
+    Start(CheatArguments<T>),
+    Stop(ContractAddress),
+    StopGlobal,
+    #[default]
+    Retain,
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct TxInfoMock {
+    pub version: CheatStatus<Felt252>,
+    pub account_contract_address: CheatStatus<Felt252>,
+    pub max_fee: CheatStatus<Felt252>,
+    pub signature: CheatStatus<Vec<Felt252>>,
+    pub transaction_hash: CheatStatus<Felt252>,
+    pub chain_id: CheatStatus<Felt252>,
+    pub nonce: CheatStatus<Felt252>,
+    pub resource_bounds: CheatStatus<Vec<Felt252>>,
+    pub tip: CheatStatus<Felt252>,
+    pub paymaster_data: CheatStatus<Vec<Felt252>>,
+    pub nonce_data_availability_mode: CheatStatus<Felt252>,
+    pub fee_data_availability_mode: CheatStatus<Felt252>,
+    pub account_deployment_data: CheatStatus<Vec<Felt252>>,
+}
+
+impl TxInfoMock {
+    #[must_use]
+    pub fn is_mocked(&self) -> bool {
+        self != &Default::default()
+    }
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct BlockInfoMock {
+    pub block_number: CheatStatus<u64>,
+    pub block_timestamp: CheatStatus<u64>,
+    pub sequencer_address: CheatStatus<ContractAddress>,
+}
+
+impl BlockInfoMock {
+    #[must_use]
+    pub fn is_mocked(&self) -> bool {
+        self != &Default::default()
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct ExecutionInfoMock {
+    pub block_info: BlockInfoMock,
+    pub tx_info: TxInfoMock,
+    pub caller_address: CheatStatus<ContractAddress>,
+    pub contract_address: CheatStatus<ContractAddress>,
+    pub entry_point_selector: CheatStatus<EntryPointSelector>,
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct TxInfoDto {
```

Comment:
DTO doesn't say really what it's for

Reply:
I don't understand. It will only be used only in communication `cairo -> forge` so DTO says exactly what it's for

Reply:
It's not clear for the reader that it's the purpose of this object. I would name it differently, something like `TxInfoMockOperations` or something like that. New terminology here is more confusing than reuse of existing terms.